### DISCUSSION TITLE
Run a script via trigger when a new custom object is being created

### DISCUSSION THREAD

**Discussion by RolandvonBorstel** [sadieblue/agilityblue.workspace-scripts#3] [Q&A] [answered]
I am trying to create a script that gets executed when a new custom object entry gets created. I can select the **Custom Object Entry** value as object for the trigger. But how do I specify a object or will the scripty then be executed in every custom object?

How can I retrieve the Object ID of the newly created object if the script gets executed for every custom object?

**Comment by bmward80** [comment]
Hi @RolandvonBorstel,

There is a command named `Get-InboundInstanceId` that you can use that contains the custom object id and the entry id, separated by a colon (:). You can use PowerShell's split utility to extract the object id from this value and then check it to make sure you'll continue operating on the object you want that script to operate on.

Here is an example:

```powershell
# This is the object id for a custom custodian object, for example
$custodianObjectId = 14

# Get the inbound object id. For custom object entries, it is in this format: "[object_id]:[entry_id]"
$inboundObjectId = Get-InboundObjectId

# Split the string and get the Object Id
$objectId = $inboundObjectId -split ":" | Select-Object -First 1

if ($objectId -ne $custodianObjectId) {
    Write-Output "Inbound entry $inboundObjectId doesn't match the custodian object."

    # The event is coming from a different object.
    # Return the inbound object instance back to the pipeline so no error is encountered for
    # "before save" events
    return Get-InboundObjectInstance
}

# The event is coming from the custom custodian object. Process what you want and return
# It back to the pipeline
$custodian = Get-InboundObjectInstance

Write-Output "Custodian Id: $($custodian.Id)"
Write-Output ""
Write-Output "JSON Representation:"
$custodian | ConvertTo-Json -Depth 10

return $custodian
```

This is a snippit from the [Concepts: Custom Objects](https://help.agilityblue.com/docs/scripting-tutorials-custom-objects#object-entries-attached-to-event-triggers) that we just updated. Check it out for more details and let me know if you need anything else.

Thanks!
Brandon

**Reply by RolandvonBorstel** [reply]
Thanks a lot for the code. If I understand it right, the Get-InboundObjectInstance command will get me a reference to the new object that has just been created, right? In the code sample it will be the custodian object.

Is there a way to get the object where the create command has been initiated in? In my case if I hit the add button in the volume object to create a new load result entry, can I get a reference to this volume entry to populate the reference field in the new load result entry?
![2024-07-10 11_26_57-Clipboard](https://github.com/sadieblue/agilityblue.workspace-scripts/assets/175238821/c7d79352-f801-472b-9411-1da9dc6cf20a)

**Comment by bmward80** [comment, marked as answer]
That's correct! the `Get-InboundObjectInstance` command will give you the instance of whatever object it is your script is listening to. So in this case, you will receive the custom object instance because you're listening to the custom object entry event.

For your second question: that level of detail isn't provided by the app, so it's not known to the script exactly where it was created from (the script is actually triggered by calls to the API). It is only aware that "something was created" and "here is that something".

That's an interesting question, though. I wonder if you could get what you need by latching onto the update volume event instead and then peek into the Loading Results reference object, retrieve that object, and check if it has a reference back to the volume? I assume you're trying to make a 2-way referenced link or something similar?

Brandon

**Reply by RolandvonBorstel** [reply]
Hi Brandon,

Thanks for your insight. The cross-reference between both objects is exactly what we want to implement. Unless there is any other simple way to see in the custom object from where it is referenced from.

I will look into your suggestion to attack the problem from the volume object perspective. 

Thanks
 
Roland

**Reply by RolandvonBorstel** [reply]
I have now been able to write a script to do what I was looking for. Thanks for the help Brandon

**Reply by bmward80** [reply]
Excellent, glad to help! 👍
